test/functional/outputs/forward_to_cloudwatch_test.go

```diff
@@ -0,0 +1,191 @@
+			b.AddSecretVolume("cloudwatch", "cloudwatch").
+				GetContainer(constants.FluentdName).
+				AddVolumeMount("cloudwatch", "/var/run/ocp-collector/secrets/cloudwatch", "", true)
+			return nil
+		}
+
+		cwlClient *cwl.Client
+
+		maxDuration, _          = time.ParseDuration("10m")
+		defaultRetryInterval, _ = time.ParseDuration("10s")
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+
+		log.V(2).Info("Creating service moto")
+		service := runtime.NewService(framework.Namespace, "moto")
+		runtime.NewServiceBuilder(service).
+			AddServicePort(5000, 5000).
+			WithSelector(map[string]string{"testname": "functional"})
+		if err := framework.Test.Client.Create(service); err != nil {
+			panic(err)
+		}
+		route := runtime.NewRoute(framework.Namespace, "moto", "moto", "5000")
+		route.Spec.TLS = &openshiftv1.TLSConfig{
+			Termination:                   openshiftv1.TLSTerminationPassthrough,
+			InsecureEdgeTerminationPolicy: openshiftv1.InsecureEdgeTerminationPolicyNone,
+		}
+		if err := framework.Test.Client.Create(route); err != nil {
+			panic(err)
+		}
+
+		cwlClient = cwl.NewFromConfig(
+			aws.Config{
+				Region: "us-east-1",
+				Credentials: aws.CredentialsProviderFunc(func(context.Context) (aws.Credentials, error) {
+					return aws.Credentials{
+						AccessKeyID:     awsAccessKeyID,
+						SecretAccessKey: awsSecretAccessKey,
+					}, nil
+				}),
+				HTTPClient: &http.Client{
+					Transport: &http.Transport{
+						TLSClientConfig: &tls.Config{
+							InsecureSkipVerify: true, //nolint:gosec
+						},
+					},
+				},
+				EndpointResolver: aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+					return aws.Endpoint{
+						PartitionID:   "aws",
+						URL:           "https://" + route.Spec.Host,
+						SigningRegion: "us-east-1",
+					}, nil
+				}),
+			})
+
+		log.V(2).Info("Creating secret cloudwatch with AWS example credentials")
+		secret := runtime.NewSecret(framework.Namespace, "cloudwatch",
+			map[string][]byte{
+				"aws_access_key_id":     []byte(awsAccessKeyID),
+				"aws_secret_access_key": []byte(awsSecretAccessKey),
+			},
+		)
+		if err := framework.Test.Client.Create(secret); err != nil {
+			panic(err)
+		}
+
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToCloudwatchOutput()
+
+		Expect(framework.DeployWithVisitors([]runtime.PodBuilderVisitor{
+			addMotoContainerVisitor,
+			mountCloudwatchSecretVisitor,
+		})).To(BeNil())
+
+		Expect(framework.WritesNApplicationLogsOfSize(numOfLogs, logSize)).To(BeNil())
+	})
+
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	Context("When sending a sequence of log messages to CloudWatch", func() {
+		FIt("should get there", func() {
+
+			var logGroupsOutput *cwl.DescribeLogGroupsOutput
+			err := wait.PollImmediate(defaultRetryInterval, maxDuration, func() (done bool, err error) {
```

Comment:
Any reason not to encapsulate this block in a similar framework method like https://github.com/openshift/cluster-logging-operator/blob/master/test/functional/outputs/forward_to_syslog_test.go#L94

Reply:
This code is specific to CloudWatch and this is the only place it's used, so I didn't think about moving it to the framework. We can do that when we need to call it from somewhere else.